src/globus_sdk/services/transfer/data/delete_data.py

```diff
@@ -66,14 +71,14 @@ class DeleteData(utils.PayloadWrapper):
 
     def __init__(
         self,
-        transfer_client,
-        endpoint,
-        label=None,
-        submission_id=None,
-        recursive=False,
-        deadline=None,
+        transfer_client: "TransferClient",
+        endpoint: str,
+        label: Optional[str] = None,
+        submission_id: Optional[str] = None,
+        recursive: bool = False,
+        deadline: Optional[Union[str, datetime.datetime]] = None,
```

Comment:
Seeing this made me suspicious of the call to `str()` below. Somewhat unrelated to typing, but I think we should we swap that to a call to `strftime()` if we're taking in `datetime` objects, so that we strictly control the format.

I'm not sure if `str(datetime)` guarantees us a good value.